sdk/java-sdk/src/main/java/kalix/javasdk/Kalix.java

```diff
@@ -119,31 +117,20 @@ public Kalix registerAction(
         Descriptors.FileDescriptor... additionalDescriptors) {
 
       final AnySupport anySupport = newAnySupport(additionalDescriptors);
-
       ActionFactory resolvedActionFactory =
           new ResolvedActionFactory(actionFactory, anySupport.resolveServiceDescriptor(descriptor));
 
-      ActionService service =
-          new ActionService(resolvedActionFactory, descriptor, additionalDescriptors, anySupport);
-
-      services.put(descriptor.getFullName(), system -> service);
-
-      return Kalix.this;
+      return registerAction(resolvedActionFactory, anySupport, descriptor, additionalDescriptors);
```

Comment:
I think we only need a `ResolvedActionFactory` when using protobuf. The Spring SDK won't need this.